QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,839 @@
+            }
+
+            var (symbol, _, _) = item;
+
+            // Convert the timestamp to exchange timezone and pass into algorithm
+            var time = GetRealTimeTickTime(exanteFeedTrade.Date, symbol);
+
+            var instrument = Client.GetSymbol(symbolId);
+
+            var size = exanteFeedTrade.Size ?? 0m;
+            var price = exanteFeedTrade.Price ?? 0m;
+            var tick = new Tick(time, symbol, "", instrument.Exchange, size, price);
+            return tick;
+        }
+
+        /// <summary>
+        /// Returns a timestamp for a tick converted to the exchange time zone
+        /// </summary>
+        private DateTime GetRealTimeTickTime(DateTime time, Symbol symbol)
+        {
+            DateTimeZone exchangeTimeZone;
+            if (!_symbolExchangeTimeZones.TryGetValue(symbol, out exchangeTimeZone))
+            {
+                // read the exchange time zone from market-hours-database
+                exchangeTimeZone = MarketHoursDatabase.FromDataFolder()
+                    .GetExchangeHours(symbol.ID.Market, symbol, symbol.SecurityType).TimeZone;
+                _symbolExchangeTimeZones.Add(symbol, exchangeTimeZone);
+            }
+
+            return time.ConvertFromUtc(exchangeTimeZone);
+        }
+
+        /// <summary>
+        /// Create a tick from the Exante tick shorts stream data
+        /// </summary>
+        /// <param name="exanteTickShort">Exante tick short stream data object</param>
+        /// <returns>LEAN Tick object</returns>
+        private Tick CreateTick(ExanteTickShort exanteTickShort)
+        {
+            if (!_subscribedTickers.TryGetValue(exanteTickShort.SymbolId, out var item))
+            {
+                // Not subscribed to this symbol.
+                return null;
+            }
+
+            var (symbol, _, _) = item;
+
+            var time = GetRealTimeTickTime(exanteTickShort.Date, symbol);
+            var bids = exanteTickShort.Bid.ToList();
+            var asks = exanteTickShort.Ask.ToList();
+            return new Tick(time, symbol, "", "",
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Size,
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Price,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Size,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Price);
+        }
+
+        #endregion
+
+        private void OnUserMessage(ExanteOrder exanteOrder)
+        {
+            Order order;
+            if (!_orderMap.TryGetValue(exanteOrder.OrderId, out order))
+            {
+                return;
+            }
+
+            var exanteSymbol = Client.GetSymbol(exanteOrder.OrderParameters.SymbolId);
+            var symbol = ConvertSymbol(exanteSymbol);
+
+            var security = _securityProvider.GetSecurity(symbol);
+            var orderFeeParameters =  new OrderFeeParameters(security, order);
+            var fee = _feeModel.GetOrderFee(orderFeeParameters);
```

Comment:
I tried to apply `security.FeeModel` and found the problems as follows. 

1. I found that I initialised `_securityProvider` with `QuantConnect.Tests.Brokerages.SecurityProvider` from tests. The class from the production is `SecurityPortfolioManager`. But it seems like overkill to create it. What's the proper way to get security from symbol?

2. Speaking of `security.FeeModel`. I check all the usages of [FeeModel setter](https://github.com/QuantConnect/Lean/blob/83f9499b4a8faf3a1344b6f22284a35ba91d2e1b/Common/Securities/Security.cs#L186-L186). And I found it fills explicitly, but not by security constructor. Why then you suggest to use `security.FeeModel` instead of class field instance `_feeModel: FeeModel`?

Reply:
> What's the proper way to get security from symbol?

There are plenty of example in Lean code to checkout, for example https://github.com/QuantConnect/Lean/blob/master/Tests/Brokerages/Kraken/KrakenFeeModelTests.cs#L38

> Why then you suggest to use security.FeeModel instead of class field instance _feeModel: FeeModel?

Because users can overwrite it and use what ever feel model they want, depending on the asset for example, shouldn't be a global fee model

Reply:
I see that `new Security` constructor at the line you pointed creates `Security` with `InteractiveBrokersFeeModel`. Also, `KrakenFeeModelTests` is a test. I'm still not following how to change `FeeModel`. I was searching for a general security provider like at 3deba219ffd5d62daba949d022726d028f575781 which provides `ExanteFeeModel`.

Reply:
Can follow example at https://github.com/QuantConnect/Lean/blob/30d7fb042bbd1da407c829efc4dbe936b282793a/Brokerages/Oanda/OandaBrokerage.cs#L67 & https://github.com/QuantConnect/Lean/blob/30d7fb042bbd1da407c829efc4dbe936b282793a/Brokerages/Oanda/OandaBrokerageFactory.cs#L98 the factory can pass in the security provider and the brokerage just needs to ask for the security, with the security the correct fee model to use can be recovered.
In the SetJob use case, like https://github.com/QuantConnect/Lean/blob/30d7fb042bbd1da407c829efc4dbe936b282793a/Brokerages/Oanda/OandaBrokerage.cs#L263 , null is passed in because the brokerage is used as a IDataQueueHandler, not like a brokerage for trading, so it should never be used in those code paths

Reply:
OK. I attempted to fix as you told at f868937345c5d226925281813349981df6a11af8